src/components/widgets/BaseInput.js

```diff
@@ -24,7 +25,8 @@ function BaseInput(props) {
       readOnly={readonly}
       autoFocus={autofocus}
       value={typeof value === "undefined" ? "" : value}
-      onChange={_onChange} />
+      onChange={_onChange}
+      onBlur={onBlur ? (event) => onBlur(inputProps.id, event.target.value) : undefined}/>
```

Comment:
I realize we could just do `onBlur && (event) => onBlur(inputProps.id, event.target.value)` here, but that may be harder to read. Opinions?